x-pack/platform/plugins/shared/cloud/public/plugin.test.ts

```diff
@@ -199,6 +199,45 @@ describe('Cloud Plugin', () => {
         const result = await setup.fetchElasticsearchConfig();
         expect(result).toEqual({ elasticsearchUrl: 'elasticsearch-url' });
       });
+      describe('exposes getInTrial', () => {
+        it('is `true` when `serverless.in_trial` is set', () => {
+          const { setup } = setupPlugin({
+            serverless: {
+              project_id: 'my-awesome-project',
+              in_trial: true,
+            },
+          });
+
+          expect(setup.getInTrial()).toBe(true);
+        });
+        it('is `false` when `serverless.in_trial` is set to false', () => {
+          const { setup } = setupPlugin({
+            serverless: {
+              project_id: 'my-awesome-project',
+              in_trial: false,
+            },
+          });
+          expect(setup.getInTrial()).toBe(false);
+        });
+        it('is `true` when `trial_end_date` is set and is in the future', () => {
+          const { setup } = setupPlugin({
+            trial_end_date: new Date(Date.now() + 10000).toISOString(),
+          });
+
+          expect(setup.getInTrial()).toBe(true);
+        });
+        it('is `false` when `trial_end_date` is set and is in the past', () => {
+          const { setup } = setupPlugin({
+            trial_end_date: new Date(Date.now() - 10000).toISOString(),
+          });
+
+          expect(setup.getInTrial()).toBe(false);
+        });
+        it('is `false` when `serverless.in_trial` & `trial_end_date` are not set', () => {
```

Comment:
What happens when both `serverless.in_trial` and `trial_end_date` are set? IMO we should throw and remove all `serverless.in_trial` test cases from this set of tests and scope those to the serverless:true config tests below.